experimental/dds/tree2/src/feature-libraries/sequence-field/compose.ts

```diff
@@ -173,6 +181,20 @@ function composeMarks<TNodeChange>(
 		composeChild,
 	);
 
+	const baseMarkIsTransient = markIsTransient(baseMark);
+	const newMarkIsTransient = markIsTransient(newMark);
+	if (baseMarkIsTransient || newMarkIsTransient) {
+		if (newMarkIsTransient) {
+			return withNodeChange(baseMark, nodeChange);
+		}
+		// TODO: Make `withNodeChange` preserve type information so we don't need to cast here
+		const nonTransient = withNodeChange(baseMark, nodeChange) as
+			| Insert<TNodeChange>
+			| Revive<TNodeChange>;
+		delete nonTransient.detachedBy;
+		return nonTransient;
```

Comment:
This seems to make assumptions about what kinds of marks `newMark` could be, and I think we should make those explicit. Our format allows composing a transient insert with a modify to the inserted node, which would behave incorrectly here.